ssl/ssl_local.h

```diff
@@ -836,6 +836,29 @@ typedef struct tls_group_info_st {
     char is_kem;             /* Mode for this Group: 0 is KEX, 1 is KEM */
 } TLS_GROUP_INFO;
 
+typedef struct tls_sigalg_info_st {
+    char *tlsname;           /* Algorithm Name as in TLS specs */
+    char *realname;          /* Algorithm Name according to provider */
+    char *algorithm;         /* Algorithm name to fetch */
+    char *oid;               /* OID of algorithm */
+    char *hash_algorithm;    /* Name of hash algorithm if any */
+    uint16_t code_point;     /* Code point of algorithm */
+    unsigned int secbits;    /* Bits of security (from SP800-57) */
+    int mintls;              /* Minimum TLS version, -1 unsupported */
+    int maxtls;              /* Maximum TLS version (or 0 for undefined) */
+    int mindtls;             /* Minimum DTLS version, -1 unsupported */
+    int maxdtls;             /* Maximum DTLS version (or 0 for undefined) */
+} TLS_SIGALG_INFO;
+
+/*
+ * Structure containing table entry of certificate info corresponding to
+ * CERT_PKEY entries
+ */
+typedef struct {
+    int nid; /* NID of public key algorithm */
```

Comment:
Should this change? Provider based algorithms don't have nids any more. Doesn't having this here impose an additional requirement on the provider to register the nid? Or does that requirement exist anyway in order for the X509 code to recognise the signature algorithms???

Reply:
Yes, that's how I see it: The (NID-using) method [OBJ_nid2obj](https://www.openssl.org/docs/man3.0/man3/OBJ_nid2obj.html) is core to the correct operation of the X.509 logic. Documentation was suitably updated in 0ac814f8247d1a5f157eec2b62bce0000c3a2d1e to make it clear that [core_obj_create](https://www.openssl.org/docs/man3.0/man7/provider-base.html) must be called (why otherwise does this method exist?).